booster/library/Booster/SMT/Interface.hs

```diff
@@ -386,32 +383,17 @@ checkPredicates ctxt givenPs givenSubst psToCheck
                     hsep ("Predicates to check:" : map (pretty' @mods) (Set.toList psToCheck))
         result <- interactWithSolver smtGiven sexprsToCheck
         case result of
-            (Unsat, Unsat) -> pure Nothing -- defensive choice for inconsistent ground truth
+            (Unsat, Unsat) -> pure $ IsUnknown Nothing -- defensive choice for inconsistent ground truth
             (Sat, Sat) -> do
                 Log.logMessage ("Implication not determined" :: Text)
-                pure Nothing
-            (Sat, Unsat) -> pure . Just $ True
-            (Unsat, Sat) -> pure . Just $ False
-            (Unknown reason, _) -> retry smtGiven sexprsToCheck transState reason
-            (_, Unknown reason) -> retry smtGiven sexprsToCheck transState reason
+                pure $ IsUnknown Nothing
+            (Sat, Unsat) -> pure $ IsSat ()
```

Comment:
We'd want to return `IsValid` rather than `IsSat` here. I think warrants a new datatype.